new/target_mode.md

```diff
@@ -0,0 +1,53 @@
+---
+RFC: unassigned
+Title: Chef Target Mode
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Chef Target Mode
+
+When deploying applications or systems, we frequently need to configure
+appliances where Chef cannot run natively. To allow us to do so, we need
+to run chef on a dedicated management node and interact with the
+appliance remotely, via an API.
+
+## Motivation
+
+    As a site reliability engineer,
+    I want to configure routers, switches and load balancers,
+    so that my application functions correctly.
+
+## Specification
+
+The concept of a node in chef is a 1:1 mapping with a client, enabling a
+single node to have several identities if needed. We can leverage this
+facility to create a target mode, allowing a management node to run chef
+for many appliances.
+Target mode is enabled by several new configuration options:
+ * `target`: The details for connectivity to the target
+   system. The result of this configuration option should be a
+   class instance that cookbooks can send messages to. This
+   enables target mode, disabling ohai entirely.
+ * `platform`: The platform that we're connecting to. Since we don't
+   have ohai, this can be used to make decisions in a cookbook.
```

Comment:
Can we do this simply by shipping an ohai plugin in a cookbook which runs first and populates the platform + friends?  I don't know what happens if we don't have a platform in between when ohai runs and early in the cookbook_loader, but if we just debug that so that it doesn't explode, then we could let this be the responsibility of the cookbook author.  All the ProviderResolver resolution is lazy, we don't need a platform at class loading time with resources or providers in order to setup the NodeMap, so I don't think there's anything preventing us from letting the platform get set later.  It might not work without some debugging but could probably be treated as bugs and just fixed.